api/swagger.yaml

```diff
@@ -181,132 +177,148 @@ paths:
               schema:
                 $ref: "#/components/schemas/Reports"
         "403":
-          description: Not authorized
+          description: Please login again
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
+
 components:
   schemas:
     Command:
       required:
       - type
-      type: object
       properties:
         type:
           type: string
-      discriminator:
-        propertyName: Type_
-      oneOf:
-      - $ref: '#/components/schemas/MoveCommand'
-      - $ref: '#/components/schemas/StopCommand'
-      - $ref: '#/components/schemas/ConstructCommand'
-      - $ref: '#/components/schemas/DecommissionCommand'
-      - $ref: '#/components/schemas/AttackCommand'
-      - $ref: '#/components/schemas/TradeCommand'
-      - $ref: '#/components/schemas/RenameCommand'
+        target:
+          type: integer
+          format: int64
+          nullable: true
+        destination:
+          $ref: "#/components/schemas/Destination"
+        resource:
+          type: integer
+          format: int64
+          nullable: true
+        amount:
+          type: integer
+          format: int64
+          nullable: true
+        ship-class:
+          type: integer
+          format: int64
+          nullable: true
+        name:
+          type: string
+          nullable: true
+
     MoveCommand:
-      allOf:
-      - $ref: '#/components/schemas/Command'
-      - required:
-        - where
-        type: object
-        properties:
-          destination:
-            $ref: "#/components/schemas/Destination"
+      type: object
+      properties:
+        destination:
+          $ref: "#/components/schemas/Destination"
+      required:
+        - destination
+
     StopCommand:
-      $ref: '#/components/schemas/Command'
+      type: object
 
     ConstructCommand:
-      allOf:
-      - $ref: '#/components/schemas/Command'
-      - required:
-        - class
-        type: object
-        properties:
-          class:
-            type: string
-          name:
-            type: string
+      type: object
+      properties:
+        ship-class:
+          type: integer
+          format: int64
+          nullable: true
```

Comment:
eg. this was optional in the generic Command, but is not optional in the ConstructCommand

Reply:
Yes, that's the point of having specific commands. In the generic one, we don't know which type is it, which means, which fields need to be filled. In the specific command, it's not optional, so we can validate on an already known type of command.

Reply:
i think you should not allow to have instances with potentially invalid data in the first place.
do the validation before you move the data into the specific structures.
that way, it shall not be nullable in the specific commands.

Reply:
If we decided to validate the generic command, we would have to write our own validator. With this approach, we can use already created validator for each of them - e.g. https://github.com/gdg-garage/space-tycoon/pull/57/files#diff-6d5cea0cd2f8a82b9b9ff66c9d77ff6c720581fae196111299e2b30d8c855ebaL21